core/src/main/scala/kafka/server/metadata/PartitionMetadataProcessor.scala

```diff
@@ -666,13 +668,15 @@ class PartitionMetadataProcessor(kafkaConfig: KafkaConfig,
         // TODO: ignore offline replicas until we support the JBOD disk failure use case
       )
       val brokerId = kafkaConfig.brokerId
-      if (partition.removingReplicas().contains(brokerId)) {
+      if (partition.removingReplicas != null &&partition.removingReplicas.contains(brokerId)) {
+        // TODO: Not sure that this makes sense... a removing replica is still a replica
```

Comment:
@rondagostino Can you take a look at this? The "removing replicas" remain valid until the reassignment completes. The broker should wait until it has been removed from the replica set before stopping replication.

Reply:
I added a commit to address this.